pkg/storage/cacher.go

```diff
@@ -531,17 +531,23 @@ func (c *Cacher) dispatchEvents() {
 func (c *Cacher) dispatchEvent(event *watchCacheEvent) {
 	triggerValues, supported := c.triggerValues(event)
 
+	// TODO: For now we assume we have a given <timeout> budget for dispatching
+	// a single event. We should consider changing to the approach with:
+	// - budget has upper bound at <max_timeout>
+	// - we add <portion> to current timeout every second
+	timeout := time.Duration(250) * time.Millisecond
```

Comment:
Yup - in large clusters it can be even hundreds (e.g. in 2k-node clusters, it can be 250 or so).
But as I wrote somewhere, it happens that some watcher sometimes block it for tens of milliseconds (e.g. 20 or 30) and system still works perfectly fine. Cancelling them would potentially cause relists, which might be even worse for the system.
Maybe 250ms is too large, but I think we shouldn't get lower than 100ms (at least now).